spring-kafka/src/main/java/org/springframework/kafka/support/DefaultKafkaHeaderMapper.java

```diff
@@ -286,20 +290,19 @@ public void fromHeaders(MessageHeaders headers, Headers target) {
 						}
 						if (!encodeToJson && valueToAdd instanceof String) {
 							target.add(new RecordHeader(key, ((String) valueToAdd).getBytes(getCharset())));
-							className = JAVA_LANG_STRING;
```

Comment:
seems the previous logic has ensured the className is assigned JAVA_LANG_STRING if valueToAdd has been String.